xdem/misc.py

```diff
@@ -23,12 +30,76 @@ def generate_random_field(shape: tuple[int, int], corr_size: int) -> np.ndarray:
     """
     field = cv2.resize(
         cv2.GaussianBlur(
-            np.repeat(np.repeat(
-                np.random.randint(0, 255, (shape[0]//corr_size,
-                                           shape[1]//corr_size), dtype='uint8'),
-                corr_size, axis=0), corr_size, axis=1),
-            ksize=(2*corr_size + 1, 2*corr_size + 1),
-            sigmaX=corr_size) / 255,
-        dsize=(shape[1], shape[0])
+            np.repeat(
+                np.repeat(
+                    np.random.randint(0, 255, (shape[0] // corr_size, shape[1] // corr_size), dtype="uint8"),
+                    corr_size,
+                    axis=0,
+                ),
+                corr_size,
+                axis=1,
+            ),
+            ksize=(2 * corr_size + 1, 2 * corr_size + 1),
+            sigmaX=corr_size,
+        )
+        / 255,
+        dsize=(shape[1], shape[0]),
     )
     return field
+
+
+def deprecate(removal_version: str | None = None, details: str | None = None):
```

Comment:
very nice addition

Reply:
Fancy!

Reply:
And including a test for deprecating functions is even fancier !!